src/store/token/actions.js

```diff
@@ -203,29 +204,86 @@ export function requestUBIPayout(payout) {
   };
 }
 
-async function loopTransfer(from, to, value, paymentNote) {
-  return await waitAndRetryOnFail(
-    () => {
-      return core.token.transfer(from, to, value, paymentNote);
-    },
-    () => {
-      return true;
-    },
-    {},
-    () => {
-      return core.token.updateTransferSteps(from, to, value);
-    },
-  );
+// Recursive helper function for transfer
+async function loopTransfer(
+  from,
+  to,
+  value,
+  paymentNote,
+  hops,
+  attemptsLeft,
+  errorsMessages = '',
+) {
+  if (attemptsLeft === 0 || hops === 0) {
+    // ran out of attempts or hops, cannot attempt further transfers
+    throw new TransferError(errorsMessages);
```

Comment:
should we here display message for users?

Reply:
If you look where this error is caught (i think in SendConfirm) it will always display something like failed transaction.  I kept these as a debug log as the error will go in the logs

Reply:
"I will also add some kind of interpretation to the user here"
Is it done somewhere, I do not see?

Reply:
Would it be ok to do this as part of the improve send confirm task?

Reply:
ok